app/src/main/cpp/skyline/services/nvdrv/devices/nvhost/as_gpu.cpp

```diff
@@ -0,0 +1,366 @@
+// SPDX-License-Identifier: MIT OR MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include <common/address_space.inc>
+#include <soc.h>
+#include <services/nvdrv/devices/deserialisation/deserialisation.h>
+#include "as_gpu.h"
+
+namespace skyline {
+    template class FlatAddressSpaceMap<u32, 0, bool, false, false, 32>;
+    template class FlatAllocator<u32, 0, 32>;
+}
+
+namespace skyline::service::nvdrv::device::nvhost {
+    using GMMU = soc::gm20b::GM20B::GMMU;
+
+    AsGpu::AsGpu(const DeviceState &state, Core &core, const SessionContext &ctx) : NvDevice(state, core, ctx) {}
+
+    PosixResult AsGpu::BindChannel(In<FileDescriptor> channelFd) {
+        // TODO: support once multiple address spaces are supported
+        return PosixResult::Success;
+    }
+
+    PosixResult AsGpu::AllocSpace(In<u32> pages, In<u32> pageSize, In<MappingFlags> flags, InOut<u64> offset) {
+        state.logger->Debug("pages: 0x{:X}, pageSize: 0x{:X}, flags: ( fixed: {}, sparse: {} ), offset: 0x{:X}",
```

Comment:
Your logs look like the Khronos Vulkan Validation Layer's ones with the space around the brackets :stuck_out_tongue: 

(I'm not asking you to change them, just something I noticed)

Reply:
Yeah I tried to do the best way to print the values in a usable format